### DISCUSSION TITLE
Is SQLAlchemy 2.0 ORM confusing?

### DISCUSSION THREAD

**Discussion by seperman** [sqlalchemy/sqlalchemy#10132] [Usage Questions]
Hello and thank you for making SQLAlchemy,
I am sure there were good reasons behind creating the SQLAlchemy 2.0 ORM, but I find the SQLAlchemy 2.0 very confusing.

It makes things more complicated compared to the v1. Except for a few examples, such as the new `session.get(Model, id)` everything else is over complicated and verbose.

I still can't wrap my head around whether `scalars` is needed.

Let's look at the examples posted in your migration guide:

| v1 | v2 |
|----|---|
| `session.query(User).all()` | `session.execute(select(User)).scalars().all()` |
|`session.query(User).filter_by(name="some user").one()` | `session.execute(select(User).filter_by(name="some user")).scalar_one()` |
|`session.query(User).filter_by(name="some user").first()`|`session.scalars(select(User).filter_by(name="some user").limit(1)).first()`|

As a long time user of SQLAlchemy, I find it painful to use the v2 API.

**Comment by zzzeek** [comment]
Hi - 

My apologies , I have no idea why we had a category called "polls" here, not sure if GitHub just added that.

This forum is for user assistance in using SQLAlchemy.  If you want to make vague polls, that's fine but I think mastodon or Twitter is more appropriate for that.

**Comment by CaselIT** [comment]
> I still can't wrap my head around whether scalars is needed.

The easiest way to reason about this is: execute always returns a sequence of tuples, regardless of the number of elements in a select.
If you have a single element, like `select(User)` or `select(User.id)`, you likely want just a list of the first item of each tuple. That's what scalars does
